### ISSUE TITLE
Use of uninitialized value in invalid_algo check results in segfaults with invalid algorithm names

### ISSUE THREAD

**Issue by jmesmon** [fireice-uk/xmr-stak#1906]
Please provide as much as possible information to reproduce the issue.

# Reproduce
1. change the pools.txt to have `"currency" : "not-a-real-coin",` (I had `"monero7"`)
2. ./xmr-stak
3. segfault (fault does not appear to occur until some work would take place using the unknown algo)

or under valgrind, the following output is produced and a warning issued about using uninitialized data:
```
$ valgrind xmr-stak 
==12717== Memcheck, a memory error detector
==12717== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==12717== Using Valgrind-3.14.0.GIT and LibVEX; rerun with -h for copyright info
==12717== Command: xmr-stak
==12717== 
==12717== Conditional jump or move depends on uninitialised value(s)
==12717==    at 0x1B81E8: jconf::parse_config(char const*, char const*) (jconf.cpp:619)
==12717==    by 0x113702: main (cli-miner.cpp:732)
==12717== 
[2018-10-11 21:43:28] : Unrecognised coin 'monero7', your options are:
        - aeon7
        - bbscoin
        - bittube
[...]
```

`jconf.cpp:619` is:
```
	if(currentCoin.GetDescription(1).GetMiningAlgo() == invalid_algo)
```

(Note: in the valgrind case we get the expected "Unrecognized" message because valgrind provides zeroed data in place of uninitialized data, which could contain any values).

# Issue with the execution
  - this is on arch linux.
  - xmr-stak --version-long
  ```
Version: xmr-stak/2.5.0/e49f444/master/lin/amd-cpu/0
  ```

**Comment by psychocrypt** [comment]
Thx for reporting. Yes we need to handle from currencies.

**Comment by psychocrypt** [comment]
closed with #1928
